src/clr-angular/data/datagrid/providers/selection.ts

```diff
@@ -251,16 +264,19 @@ export class Selection {
          * If at least one item isn't selected, we select every currently displayed item.
          */
         if (this.isAllSelected()) {
-            this.current = this.current.filter(item => {
-                return this._items.displayed.indexOf(item) < 0;
+            this._items.displayed.forEach((item, displayIndex) => {
+                const currentIndex = this.current.indexOf(this._items.trackBy(displayIndex, item));
```

Comment:
You don't want the `trackBy` here to test against `current`. This should simply be:
```typescript
const currentIndex = this.current.indexOf(item);
```

Reply:
You are correct.  I've updated the commit accordingly. 

Sorry, that was left over from an attempt at getting the index from `this.selected` instead of `this.current`.  That failed one of the other units tests. I realized that when `set current` is called, `selected` does not get updated at that time.

That might expose a separate issue, but outside of the current scope I believe.